### PR TITLE
(optionally) Save and restore the checkpoint iteration counter

### PR DESCRIPTION
**Pull request by ChrisCummins**
This patch adds a training flag `-reset_iterations` which allows users to restore the iteration count when resuming training from a checkpoint. When set to '1' (the default value), the iteration count is reset 0 when loading from a checkpoint:

    $ th train.lua $OPTIONS -init_from cv/checkpoint_20.t7
    Running in CPU mode
    Initializing from 	cv/checkpoint_20.t7
    Epoch 1.00 / 50, i = 1 / 17800, loss = 3.075188
    Epoch 1.01 / 50, i = 2 / 17800, loss = 3.014080
    ...

This is in keeping with the behaviour of upstream. However, if `-reset_iterations` is set to 0, the iteration counter (and thus epoch) is loaded from the checkpoint:

    $ th train.lua $OPTIONS -init_from cv/checkpoint_20.t7 -reset_iterations 0
    Running in CPU mode
    Initializing from 	cv/checkpoint_20.t7
    Epoch 1.06 / 50, i = 21 / 17800, loss = 3.075188
    Epoch 1.06 / 50, i = 22 / 17800, loss = 3.014080
    ...

### CODE REVIEW

**Comment by jcjohnson** [comment]
Looks good to me!
